x-pack/plugins/case/common/api/cases/status.ts

```diff
@@ -21,6 +23,11 @@ export const CaseStatusRt = rt.union([
 
 export const caseStatuses = Object.values(CaseStatuses);
 
+export const CaseStatusFilterRt = rt.union([rt.literal(AllCaseType), ...CaseStatusRt.types]);
```

Comment:
Do you think there's a way we could avoid adding `all` to the status types? If we add it here that means a case could potentially have a status of type `all` which wouldn't really make sense 🤔

Reply:
@angorayc I agree with @jonathan-buttner . It will nice if we can avoid it.

Reply:
I see how you meant. I have `CaseStatusRt` that doesn't include `all` for cases, and use `CaseStatusFilterRt` just for rendering the filter relevant stuff.
I also remove the status from api call if the filter is set to `all`. Hope this makes it a bit clearer.

Reply:
@angorayc As we discuss I removed the `CaseStatusFilterRt`.